File: binarySearch.py
# binary searchđ
def binarySearch(mylist, item):
    # initial parameters
    low = 0
    high = len(mylist) - 1
    # while loop, dokler, low ni večja od higha
    while low <= high:
        mid = (low + high) / 2
        # mid je premikajoči se index
        # guess je pa številka tega indexa
        guess = mylist[round(mid)]
        # če je guess enaki itemu, potem returnaj guess
        if guess == item:
            return round(mid)
        # če je guess večji od itema, potem je high mid -1
        if guess > item:
            high = mid - 1
        else:
            low = mid + 1
    return None


my_list = [1, 3, 5, 7, 8, 9]
print(binarySearch(my_list, 7))

File: binarySearch.js
let list = [1, 2, 3, 4, 5, 6, 7];

const bintarySearch = (list, item) => {
  let low = 0;
  let high = list.length - 1;
  while (low <= high) {
    let mid = (low + high) / 2;
    let guess = list[mid];
    if (guess === item) {
      return mid;
    } else if (guess > item) {
      high = mid - 1;
    } else if (guess < item) {
      low = mid + 1;
    } else {
      return null;
    }
  }
};

console.log(bintarySearch(list, 98));
